lisa/operating_system.py

```diff
@@ -362,7 +362,7 @@ def _install_packages(
             )
 
     def _package_exists(self, package: str, signed: bool = True) -> bool:
-        command = f"apt list --installed | grep -Ei {package}"
+        command = f"dpkg --get-selections | grep '{package}[[:space:]]*install$'"
```

Comment:
1. Don't use grep here, use Python regexp. It can output the original value, and help troubleshooting.
2. Add comments to explain the expected result here.